build(corda-all-in-one-obligation): fix image build error - unable to select packages

**Describe the bug**
In the process of the container building of corda-all-in-one-obligation, there are the following errors and the container cannot be built.

```
[process] 
DOCKER_BUILDKIT=1 docker build ./tools/docker/corda-all-in-one/ -f ./tools/docker/corda-all-in-one/corda-v4_8/Dockerfile

```

```
[error 1]
#16 [11/44] RUN apk add --no-cache openssh augeas
#16 0.311 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
#16 0.402 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
#16 0.644 ERROR: unable to select packages:
#16 0.646   openssh-client-common-9.3_p1-r3:
#16 0.646     breaks: openssh-client-default-9.3_p2-r0[openssh-client-common=9.3_p2-r0]
#16 ERROR: process "/bin/sh -c apk add --no-cache openssh augeas" did not complete successfully: exit code: 2
------
 > [11/44] RUN apk add --no-cache openssh augeas:
0.311 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
0.402 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
0.644 ERROR: unable to select packages:
0.646   openssh-client-common-9.3_p1-r3:
0.646     breaks: openssh-client-default-9.3_p2-r0[openssh-client-common=9.3_p2-r0]
------
Dockerfile:37
--------------------
  35 |     
  36 |     # OpenSSH - need to have it so we can shell in and install/instantiate contracts and troubleshoot
  37 | >>> RUN apk add --no-cache openssh augeas
  38 |     
  39 |     # Configure the OpenSSH server we just installed
--------------------
ERROR: failed to solve: process "/bin/sh -c apk add --no-cache openssh augeas" did not complete successfully: exit code: 2
Error: Process completed with exit code 1.

```

```
[error 2]
=> ERROR [29/45] RUN java -jar /corda-tools-shell-cli-all.jar install-shell-extensions                                                                                 0.8s 
------                                                                                                                                                                       
 > [29/45] RUN java -jar /corda-tools-shell-cli-all.jar install-shell-extensions:                                                                                            
0.752 Error: Invalid or corrupt jarfile /corda-tools-shell-cli-all.jar
------
Dockerfile:69
--------------------
  67 |     RUN curl https://software.r3.com/artifactory/corda-releases/net/corda/corda-tools-shell-cli/${CORDA_TOOLS_SHELL_CLI_VERSION}/corda-tools-shell-cli-${CORDA_TOOLS_SHELL_CLI_VERSION}-all.jar --output /corda-tools-shell-cli-all.jar
  68 |     # This is what makes the "corda-shell" alias avaialble on the terminal
  69 | >>> RUN java -jar /corda-tools-shell-cli-all.jar install-shell-extensions
  70 |     
  71 |     RUN git clone https://github.com/corda/samples-kotlin.git
--------------------
ERROR: failed to solve: process "/bin/sh -c java -jar /corda-tools-shell-cli-all.jar install-shell-extensions" did not complete successfully: exit code: 1
```

**To Reproduce**
[error 1]
> Run the [process] command above and it will try to build the image.

[error 2]
> modify tools/docker/corda-all-in-one/corda-v4_8/Dockerfile and replace the following:
36 # OpenSSH - need to have it so we can shell in and install/instantiate contracts and troubleshoot
37 RUN apk add --no-cache openssh augeas

With
```
RUN apk upgrade
RUN apk add --no-cache openssh=9.3_p2-r0
RUN apk add --no-cache augeas=1.12.0-r5
```

Hello @petermetz 

[error 2] has now been fixed by  this commit https://github.com/hyperledger/cacti/commit/01c52652098405673e2b0bbb72852f7ba35f5816

and we have encountered a new error when building the corda v4-8 image.

[error 3]
```

9.066 FAILURE: Build failed with an exception.
9.066 
9.066 * What went wrong:
9.066 Could not initialize class org.codehaus.groovy.reflection.ReflectionCache
9.066 > Exception java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7 [in thread "Daemon worker"]
9.066 
9.066 * Try:
9.066 Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
9.066 
9.066 * Get more help at https://help.gradle.org
9.066 
9.066 BUILD FAILED in 8s
------
Dockerfile:84
--------------------
  82 |     RUN apk add --no-cache supervisor
  83 |     
  84 | >>> RUN ./gradlew build deployNodes
  85 |     
  86 |     # OpenSSH server
--------------------
ERROR: failed to solve: process "/bin/sh -c ./gradlew build deployNodes" did not complete successfully: exit code: 1
```

--- a/tools/docker/corda-all-in-one/corda-v4_8/Dockerfile
+++ b/tools/docker/corda-all-in-one/corda-v4_8/Dockerfile
@@ -1,8 +1,8 @@
-FROM docker:24.0.2-dind
+FROM docker:24.0.5-dind
 
 # cordaVersion=4.8.5
 # cordaCoreVersion=4.8.5
-ARG SAMPLES_KOTLIN_SHA=1504878ce446555bd861bbe4dd3d1154e905a07f
+ARG SAMPLES_KOTLIN_SHA=8ce33f57ad1dd77f721af4d1f59d72f3d331fd0c
 ARG SAMPLES_KOTLIN_CORDAPP_SUB_DIR_PATH="./Advanced/obligation-cordapp/"
 ARG CORDA_TOOLS_SHELL_CLI_VERSION=4.8
 
@@ -11,7 +11,7 @@ WORKDIR /
 RUN apk update
 
 # Install dependencies of Docker Compose
-RUN apk add py-pip python3-dev libffi-dev openssl-dev gcc libc-dev make
+RUN apk add docker-cli docker-cli-compose
 
 # Install git so we can check out the kotlin-samples repo of Corda
 RUN apk add --no-cache git
@@ -25,16 +25,17 @@ RUN apk add --no-cache curl
 # The file binary is used to inspect exectubles when debugging container image issues
 RUN apk add --no-cache file
 
-RUN apk add --no-cache openjdk8
-
-# Need gradle to execute the corda sample app setup commands
-RUN apk add --no-cache gradle
+RUN apk add openjdk8
 
 ENV CACTUS_CFG_PATH=/etc/hyperledger/cactus
 RUN mkdir -p $CACTUS_CFG_PATH
 
+# Need to install augtool for manipulating the SSH configuration files below
+RUN apk add --no-cache augeas
+
 # OpenSSH - need to have it so we can shell in and install/instantiate contracts and troubleshoot
-RUN apk add --no-cache openssh augeas
+RUN apk del openssh*
+RUN apk add --no-cache openssh
 
 # Configure the OpenSSH server we just installed
 RUN augtool 'set /files/etc/ssh/sshd_config/AuthorizedKeysFile ".ssh/authorized_keys /etc/authorized_keys/%u"'
@@ -70,6 +71,9 @@ WORKDIR /samples-kotlin
 RUN git checkout ${SAMPLES_KOTLIN_SHA}
 
 WORKDIR /samples-kotlin/${SAMPLES_KOTLIN_CORDAPP_SUB_DIR_PATH}
+
+# Set the platform versionn to 11 instead of the default 12 for the basic cordapps
+RUN sed -i "s/platformVersion=12/platformVersion=11/g" /samples-kotlin/Basic/constants.properties
 
 # Install supervisord because we need to run the docker daemon and also the corda network
 # meaning that we have multiple processes to run.

